lib/spack/spack/util/executable.py

```diff
@@ -122,6 +119,10 @@ def __call__(self, *args, **kwargs):
         By default, the subprocess inherits the parent's file descriptors.
 
         """
+        # Find exe full path if in current directory
+        if Path(self.exe[0]).exists():
+            self.exe[0] = (Path.cwd() / self.exe[0]).as_posix()
```

Comment:
I'm on the fence as to whether this should be done, but if it is done, it should happen in `__init__`.

Reply:
Unfortunately, this has to be in the call. There are instances where the Executable class is instantiated from a different current working directory than it is called from.

Reply:
I worry this is possibly confusing: Spack performs multiple directory changes and it may be hard to debug if the logic moves those around. Therefore I’d prefer that what the Executable was referring to stayed consistent.

Could we always convert to an absolute path in __init__?

Reply:
We wouldn't be able to find the file in **init** because Execuatable can be created in a different dir than it will be called in. Say we have `Executable(./my_executable)` instantiated from `/my_dir/cwd1`. Spack then changes dirs to work out of `/my_dir/cwd2`. The executable class is then called. The old behavior of spack was to execute `./my_executable` which would have found `/my_dir/cwd2/my_executable`. When we're in **init**, we don't know what the future cwd will be so we don't know where to search to find the full path.

Reply:
> We wouldn't be able to find the file in init because Execuatable can be created in a different dir than it will be called in

I want this case to fail: the user should provide the appropriate directory at `__init__` time; I don't want Spack to worry about whether CWD could change such to make it available later